### ISSUE TITLE
Holopad future features

### ISSUE THREAD

**Issue by chromiumboy** [space-wizards/space-station-14#33954]
Here's a general list of features to consider for holopads in the future:

- [ ] Include a text-based filter at the top of the contacts list so users can quickly narrow down a search for specific holopads
- [ ] Use the view subscription system to preload entities into the AI player's PVS when they get a summons from a holopad
- [ ] Introduce a system to personalize AI holographic avatars
- [ ] Give holopads the ability to change the color of the holograms they project and those to which they are connected too (e.g., syndicate holopads being able to turn the holograms on linked holopads red)
- [ ] Add additional controls to the 'in-call' UI, such as a mute toggle and a conference mode (i.e., change the listening range of the holopad) toggle 
- [ ] Have speech relayed from the holopad holograms rather than the holopad unit itself
- [ ] Include the name of the holopad that is transmitting a call alongside the caller ID on the 'receiving call' UI

**Comment by spanky-spanky** [comment]
Is there any chance we could have emotes transmitted through holopads? Mime communication is very quiet otherwise.

**Comment by chromiumboy** [comment]
> Is there any chance we could have emotes transmitted through holopads? Mime communication is very quiet otherwise.

The current chat system doesn't support it, we'll have to wait until the refactor is done

**Comment by spanky-spanky** [comment]
> > Is there any chance we could have emotes transmitted through holopads? Mime communication is very quiet otherwise.
> 
> The current chat system doesn't support it unfortunately, we'll have to wait until the refactor is done

![image](https://github.com/user-attachments/assets/8069f1a2-903d-4176-a83f-e5d3057c16c5)
Interestingly enough emotes through the emotes menu (Y) work.

**Comment by ScarKy0** [comment]
> Interestingly enough emotes through the emotes menu (Y) work.

Funnily enough, thats the only type of emoting that the chat system supports that works for holopads

**Comment by chromiumboy** [comment]
> > > Is there any chance we could have emotes transmitted through holopads? Mime communication is very quiet otherwise.
> > 
> > 
> > The current chat system doesn't support it unfortunately, we'll have to wait until the refactor is done
> 
> ![image](https://private-user-images.githubusercontent.com/180730777/397550048-8069f1a2-903d-4176-a83f-e5d3057c16c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2NDk3NzQsIm5iZiI6MTczNDY0OTQ3NCwicGF0aCI6Ii8xODA3MzA3NzcvMzk3NTUwMDQ4LTgwNjlmMWEyLTkwM2QtNDE3Ni1hODNmLWU1ZDMwNTdjMTZjNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxOVQyMzA0MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZDQ0MDczNjg5ZDc4YjA1NzU0NTlhNGU1OTMyMjBjMDFhNjgwMDBlYmEzYTNiNGY2NDgyYjQ2NGZjZGI1MTJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.H7qmopzMYVX88JJ6gEQPAwJbLxdm9P5X6NULlY08RCg) Interestingly enough emotes through the emotes menu (Y) work.

Oh yes, I added support for standard emotes. I thought you meant custom emotes, my bad

**Comment by Bomjojuk** [comment]
Are there any plans to realise the "detached mode" For ai hologram similar to one that ss13 paradise has?

**Comment by Errant-4** [comment]
> Are there any plans to realise the "detached mode" For ai hologram similar to the one that ss13 paradise has?

What does that do? Allow the hologram to walk around it within range?

**Comment by ScarKy0** [comment]
> What does that do? Allow the hologram to walk around it within range?

Allows AI to walk around in a small area around the holopad

**Comment by chromiumboy** [comment]
> Are there any plans to realise the "detached mode" For ai hologram similar to the one that ss13 paradise has?

No, no plans. I'm not a fan of it. The AI already has unrestricted movement in its free roam mode. Being anchored at the holopad gives its physicality contrast

**Comment by ASTRALIXIX** [comment]
Is it possible to use the AI's general camera vision when it uses a holopad?

**Comment by chromiumboy** [comment]
> Is it possible to use the AI's general camera vision when it uses holopad in the future?

It's certaibly possible, but I don't plan on changing it

**Comment by Carolyn3114** [comment]
> > Are there any plans to realise the "detached mode" For ai hologram similar to the one that ss13 paradise has?
> 
> No, no plans. I'm not a fan of it. The AI already has unrestricted movement in its free roam mode. Being anchored at the holopad gives its physicality contrast to that freedom

it looks cool though and what you're saying is so nitpicky
